lib/kinesis/manager/Manager.rb

```diff
@@ -0,0 +1,212 @@
+          stream_name: local.name,
+          shard_count: local.shards
+        })
+
+        @client.wait_until(:stream_exists, {
+          stream_name: local.name
+        })
+
+        # Describe the newly created stream
+        created_stream = Kinesis::describe_stream(local.name)
+
+        # If the stream retention period is different, then update it
+        if created_stream.retention_period_hours > local.retention_period
+          @client.decrease_stream_retention_period({
+            stream_name: local.name,
+            retention_period_hours: local.retention_period
+          })
+        elsif created_stream.retention_period_hours < local.retention_period
+          @client.increase_stream_retention_period({
+            stream_name: local.name,
+            retention_period_hours: local.retention_period
+          })
+        end
+
+        # If the created stream has tags, add them
+        if !local.tags.empty?
+          @client.add_tags_to_stream({
+            stream_name: local.name,
+            tags: local.tags
+          })
+        end
+
+      end
+
+      def update(local, diffs)
+        diffs.each do |diff|
+          case diff.type
+          when StreamChange::SHARDS
+
+            # See if we are splitting or merging and make sure it is a multiple of 2
+            if diff.aws < diff.local
+              if diff.aws != diff.local / 2.0
+                puts Colors.red("Can only increase the number of shards by a factor of 2")
+              else
+                aws_stream = Kinesis::named_streams[local.name]
+
+                # Split the shards 1 at a time
+                aws_stream.sorted_shards.each do |shard|
+                  puts Colors.blue("Splitting shard #{shard.shard_id}")
+
+                  # The splitting point is halfway between the hash start ane end
+                  hash_start = shard.hash_key_range.starting_hash_key.to_i(10)
+                  hash_end = shard.hash_key_range.ending_hash_key.to_i(10)
+                  hash_split = hash_start + ((hash_end - hash_start) / 2)
+
+                  @client.split_shard({
+                    stream_name: local.name,
+                    shard_to_split: shard.shard_id,
+                    new_starting_hash_key: hash_split.to_s(10)
+                  })
+
+                  # After every split we have to wait until the stream is ready
+                  @client.wait_until(:stream_exists, {
+                    stream_name: local.name
+                  })
+                end
+
+              end
+            elsif diff.aws > diff.local
+              if diff.aws != diff.local * 2.0
+                puts Colors.red("Can only decrease the number of shards by a factor of 2")
+              else
+
+                aws_stream = Kinesis::named_streams[local.name]
+
+                # Merge the sorted shards in groups of 2
+                aws_stream.sorted_shards.each_slice(2) do |slice|
+
+                  if slice.length != 2
```

Comment:
Yeah this check was actually kind of unsafe anyways.  So now I'm checking that the aws stream is double the size of the local one and quitting, which will ensure each slice is 2 in length (since it would have to be an even number of shards)